x-pack/plugins/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -623,16 +623,20 @@ export const LensTopNavMenu = ({
             share.toggleShareContextMenu({
               anchorElement,
               allowEmbed: false,
-              allowShortUrl: false, // we'll manage this implicitly via the new service
-              shareableUrl: shareableUrl || '',
-              shareableUrlForSavedObject: savedObjectURL.href,
+              allowShortUrl: false,
+              delegatedShareUrlHandler: () => {
+                return shareUrlEnabled ? shareableUrl! : savedObjectURL.href;
+              },
               objectId: currentDoc?.savedObjectId,
               objectType: 'lens',
-              objectTypeTitle: i18n.translate('xpack.lens.app.share.panelTitle', {
-                defaultMessage: 'visualization',
-              }),
+              objectTypeMeta: {
+                title: i18n.translate('xpack.lens.app.shareModal.title', {
+                  defaultMessage: 'Share this Lens visualization',
+                }),
+              },
               sharingData,
-              isDirty: isCurrentStateDirty,
+              // only want to know about changes when savedObjectURL.href
+              isDirty: !shareUrlEnabled ? isCurrentStateDirty : false,
```

Comment:
Also here, I think the check is not necessary.

```suggestion
              isDirty: isCurrentStateDirty,
```

Reply:
This is necessary to confirm that the unsaved behavior isn't shown in link_content.tsx for shareUrlEnabled. It is a fix from a previous issue you had pointed out.